File: Article-Figures/Methodo/figure3_mexican_hat.r
setwd("D:\\Documents\\Figures\\Methodo")

DOG <- function(x,y, sx, sy, K, inh){ ## similar than connectionmap4() in outil.calcul.py
    A1 <- 1.0 + inh
    A2 <- inh
    M1 <- A1*exp(-(  ((x)**2)/(2*sx**2) + ((y)**2)/(2*sy**2) ) )
    M2 <- -A2*exp(-(  ((x)**2)/(2*(K*sx)**2) + ((y)**2)/(2*(K*sy)**2) ) )
    return(M1+M2)
}

## chapeau used for auto-inhibition:
x <- seq(0,30,0.1)
SA1 <- DOG(x,0,5.0,5.0,1.2,6.0)
SA2 <- DOG(x,0,5.0,5.0,2.0,1.42868)
SA3 <- DOG(x,0,5.0,5.0,1.2,8.0)
pdf("pepin.pdf")
plot(x, SA3*200, col="green", type='l', lwd=3, lty=2)
points(x, SA2*200, col="red", type='l', lwd=3, lty=5)
points(x, SA1*200, type='l', col="black", lwd=3)
dev.off()

File: Article-Figures/Methodo/figure3-mexican_hat.py
import matplotlib.pyplot as plt
import numpy as np


def MH(A, mu, sigma, K, B, x):
    g1 = (1+B)*np.exp(-((x-mu)**2)/(2*(sigma)**2)) #+ A*exp((-(x-mu-100)**2)/(sigma**2))
    g2 = B*np.exp(-((x-mu)**2)/(2*(K*sigma)**2))
    return A*(g1-g2)

x = np.arange(0, 30, 0.1)
SA1 = MH(200, 0, 5.0, 1.2, 6.0,x)
SA2 = MH(200, 0, 5.0, 2.0, 1.42868,x)
SA3 = MH(200, 0, 5.0, 1.2, 8.0,x)

plt.plot(x, SA1, "-", x, SA2, "r--", x, SA3, "g-.")
plt.show()